### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Shubhamkr2610**
Added auth feature . 
files to check : 
src/pages/loginpage/Login.js
src/pages/loginpage/login.css
src/pages/profile/Profile.js
src/pages/profile/profile.css
src/pages/signup/Signup.js
src/pages/signup/signup.css
scr/context/authContext.js
image

image
image

### CODE REVIEW

File: src/context/authContext.js [hunk 1]
```diff
@@ -9,11 +9,11 @@ const AuthProvider = ({children}) =>{
 
     const login = async (e,email,password)=> {
         e.preventDefault()
-        let userData = JSON.stringify({email,password})
+        // let userData ={email,password}
```
Comment (arjun-computer-geek) [inline src/context/authContext.js]: Remove commented code

File: src/context/authContext.js [hunk 2]
```diff
@@ -28,10 +28,33 @@ const AuthProvider = ({children}) =>{
...
+    }
+
+    const signup = async (e, userData) =>{
+        e.preventDefault()
+
+      
+        try{
+           const {data} =await axios.post("/api/auth/signup", userData)
+
+
```
Comment (arjun-computer-geek) [inline src/context/authContext.js:35]: Remove extra spaces

File: src/pages/loginpage/login.css [hunk 3]
```diff
@@ -1,9 +1,59 @@
+.login-wrapper{
+    min-height: 86vh;
+    /* border: 2px solid black; */
```
Comment (arjun-computer-geek) [inline src/pages/loginpage/login.css]: Remoce unused code

File: src/pages/profile/Profile.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import './profile.css'
+import { Avatar } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { useAuth } from '../../context/authContext'
```
Comment (arjun-computer-geek) [inline src/pages/profile/Profile.js:5]: You can use relative path insted absolute

**Review by arjun-computer-geek — APPROVED** [review]
Given some suggestions. Else code is clean. And readable

File: src/context/authContext.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    // {id: "" , enocdedToken: "", firstName: "", lastName: "" , email: ""}
+    const [user , setUser] = useState(null);
+
+    const login = async (e,email,password)=> {
+        e.preventDefault()
+        let userData = JSON.stringify({email,password})
+        // console.log(userData)
+
+        try{
+            const {data} = await axios.post("/api/auth/login", userData)
```
Comment (Rockyraven) [inline src/context/authContext.js]: remove this  unwanted code

File: src/context/authContext.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        catch(error){
+            console.log(error)
+        }
+        
+    }
+    // setUser("shu")
+    // console.log(user)
+    return(
+        <authContext.Provider value={{login, user}} >
+            {children}
```
Comment (Rockyraven) [inline src/context/authContext.js]: remove this too

**Review by Rockyraven — APPROVED** [review]
your code is clean and easy to readable you can good to go

**Pull request merged**